### PR TITLE
BUG: Series.mask with small int dtypes raising

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jreback** [comment]
can you add a whatsnew here

File: pandas/tests/series/indexing/test_setitem.py [hunk 1]
```diff
@@ -364,6 +364,48 @@ def test_setitem_nan_with_bool(self):
...
+        ser[mask] = alt
+        expected = Series([245, 2, 246], dtype="uint8")
+        tm.assert_series_equal(ser, expected)
+
+        ser2 = orig.copy()
+        ser2.mask(mask, alt, inplace=True)
+        tm.assert_series_equal(ser2, expected)
+
+        # FIXME: don't leave commented-out
+        # FIXME: ser.where(~mask, alt) unnecessarily upcasts to int64
```
Comment (mroeschke) [inline pandas/tests/series/indexing/test_setitem.py:404]: Should this be an xfail?

Comment (jbrockmendel) [reply pandas/tests/series/indexing/test_setitem.py:404]: probably.  i didnt want to make it a separate test bc im planning/hoping to move the whole thing into a SetitemCastingEquivalents subclass

Comment (mroeschke) [reply pandas/tests/series/indexing/test_setitem.py:404]: Okay fine as is then

**Review by mroeschke — APPROVED** [review]

**Comment by jbrockmendel** [comment]
> can you add a whatsnew here

whatsnew added + green

**Pull request merged**